src/testapp/sessions/orbits.cpp

```diff
@@ -0,0 +1,234 @@
+/**
+ * Open Space Program
+ * Copyright © 2019-2022 Open Space Program Project
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "orbits.h"
+#include "common.h"
+
+#include <adera/drawing/CameraController.h>
+
+#include <osp/core/math_2pow.h>
+#include <osp/drawing/drawing.h>
+#include <osp/universe/coordinates.h>
+#include <osp/universe/universe.h>
+#include <osp/util/logging.h>
+
+#include <random>
+
+#include <fenv.h>
+
+using namespace adera;
+using namespace osp::draw;
+using namespace osp::universe;
+using namespace osp;
+
+namespace testapp::scenes
+{
+
+    osp::Session setup_orbit_planets(
+        TopTaskBuilder &rBuilder,
+        ArrayView<entt::any> topData,
+        Session &uniCore,
+        Session &uniScnFrame)
+    {
+        feenableexcept(FE_INVALID | FE_OVERFLOW);
```

Comment:
this function appears to only exist on linux?

I'm not sure enabling floating point exception trapping in a platform specific manner is the right approach.

if you're doing math operations that are prone to encountering floating point exceptions, i'd rather see standardized / standard-library functionality used only, unless there's a substantial performance improvement to be had from calling out to platform specific functions.

seems like this header has the standardized stuff in it? https://en.cppreference.com/w/cpp/numeric/fenv